lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2856,8 +2764,20 @@ def detect_ipvX_address_usage():
         self.option_minimal_versions['stop_timeout']['supported'] = stop_timeout_supported
 
     def __init__(self, **kwargs):
-        super(AnsibleDockerClientContainer, self).__init__(**kwargs)
-        self._get_minimal_versions()
+        def detect_ipvX_address_usage():
+            '''
+            Helper function to detect whether any specified network uses ipv4_address or ipv6_address
+            '''
+            for network in self.module.params.get("networks") or []:
+                if network.get('ipv4_address') is not None or network.get('ipv6_address') is not None:
+                    return True
+            return False
+        if 'option_minimal_versions' in kwargs:
+            if 'ipvX_address_supported' in kwargs['option_minimal_versions']:
+                if 'detect_usage' not in kwargs['option_minimal_versions']['ipvX_address_supported']:
+                    kwargs['option_minimal_versions']['ipvX_address_supported']['detect_usage'] = detect_ipvX_address_usage
```

Comment:
I'm not a huge fan of this, but I can't think of a better way to get this function passed in to the superclass. Any ideas?

Reply:
Hmmm, this is indeed somewhat ugly.

I would simply put it into `main()`. Or at least get rid of the `if`s and simply write
```
kwargs['option_minimal_versions']['ipvX_address_supported'] = dict(docker_py_version='1.9.0', usage_msg='ipv4_address or ipv6_address in networks', detect_usage=detect_ipvX_address_usage)
```
After all, nobody else is using `AnsibleDockerClientContainer`.

Alternatively, we could define `option_minimal_versions` here and pass it to the super-constructor. This will blow up the constructor a bit, but at least it will be a bit clearer :)